src/context/TodoContext.tsx

```diff
@@ -0,0 +1,77 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import React, { ReactNode, useState, useMemo } from 'react';
+
+import { useLocalStorage } from '../hooks/useLocalStorage';
+import { Status } from '../types/Status';
+import { Todo } from '../types/Todo';
+
+type Props = {
+  children: ReactNode;
+};
+
+export const TodoContext = React.createContext({
+  todos: [] as Todo[],
+  status: Status.All,
+});
+
+export const TodoUpdateContext = React.createContext({
+  addTodo: (_todo: Todo) => { },
+  deleteTodo: (_todoId: string) => { },
+  deleteCompletedTodos: () => { },
+  updateTodo: (_todo: Todo) => { },
+  changeTodosStatus: () => { },
+  setStatus: (_status: Status) => { },
+  isTodosCompleted: false,
+});
```

Comment:
for this part of code: 
why do we need to create some callbacks like  `addTodo: (_todo: Todo) => { },`?
for me it seems, you tried add some types to your context. But you can use generic to solve it.

exmaple:
```
const noop = () => {};

interface TodosContextValues {
  addTodo: (todo: Todo) => void,
  deleteTodo: (id: string) => void,
  deleteCompletedTodos: () => void,
}

const TodoContext = React.createContext<TodosContextValues>({
  addTodo: noop,
  deleteTodo: noop,
  deleteCompletedTodos: noop,
})
```


and why do we need 2 contexts? TodoContext and TodoUpdateContext
1 context is completely enough